sources/assets/Stride.Core.Assets.Yaml/DynamicYaml/DynamicYamlObject.cs

```diff
@@ -1,50 +1,49 @@
 // Copyright (c) .NET Foundation and Contributors (https://dotnetfoundation.org/ & https://stride3d.net) and Silicon Studio Corp. (https://www.siliconstudio.co.jp)
 // Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+
 using System.Dynamic;
 using System.Globalization;
 using Stride.Core.Yaml.Serialization;
 
-namespace Stride.Core.Yaml
+namespace Stride.Core.Yaml;
+
+public abstract class DynamicYamlObject : DynamicObject
 {
-    public abstract class DynamicYamlObject : DynamicObject
+    protected static YamlNode ConvertFromDynamic(object? obj)
     {
-        protected static YamlNode ConvertFromDynamic(object obj)
-        {
-            if (obj == null)
-                return new YamlScalarNode("null");
+        if (obj == null)
+            return new YamlScalarNode("null");
 
-            if (obj is string)
-            {
-                return new YamlScalarNode((string)obj);
-            }
-
-            if (obj is YamlNode)
-                return (YamlNode)obj;
-
-            if (obj is DynamicYamlMapping)
-                return ((DynamicYamlMapping)obj).Node;
-            if (obj is DynamicYamlArray)
-                return ((DynamicYamlArray)obj).node;
-            if (obj is DynamicYamlScalar)
-                return ((DynamicYamlScalar)obj).node;
+        if (obj is string str)
```

Comment:
nit: Could switch patern matching be used here?

Reply:
It's possible. It wasn't suggested by code analysis. I'll give it a try.